proxy_test.go

```diff
@@ -98,29 +104,45 @@ func TestReverseProxyHash(t *testing.T) {
 		assert.True(t, numHits > 0, "One down - some hit")
 		assert.True(t, numMisses < n, "One down - not all missed")
 		assert.True(t, numMisses > 0, "One down - some missed")
+		assert.Equal(t, n, numHits+numMisses, "One down - hits and misses adds up")
 		accHits += numHits
 		accMisses += numMisses
 	}
 	assert.Equal(t, n, accMisses, "Misses add up")
 }
 
 func measHitMiss(t *testing.T, client *redis.Client, n int, keyTempl string) (numHits, numMisses int, err error) {
+	type tresult struct {
+		hit bool
+		err error
+	}
+	results := make(chan tresult)
 	for i := 0; i < n; i++ {
 		key := fmt.Sprintf(keyTempl, i)
-		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-		defer cancel()
-		args := client.Query(ctx, "GET", key)
-		for j := 0; j < args.Len(); j++ {
+		go func(key string, rq chan<- tresult) {
+			ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+			defer cancel()
+			args := client.Query(ctx, "GET", key)
+			assert.Equal(t, 1, args.Len(), "GET returns a single value")
 			var v string
 			args.Next(&v)
-			if v == "1" {
-				numHits++
+			hit := v == "1"
+			if err := args.Close(); err != nil {
+				rq <- tresult{false, err}
 			} else {
-				numMisses++
+				rq <- tresult{hit, nil}
 			}
+		}(key, results)
+	}
+	for i := 0; i < n; i++ {
```

Comment:
After our discussion there's another pattern than I thought that you can use here (no need to change tho):
```go
wg := &sync.WaitGroup{}

for i := 0; i != n; i++ {
    wg.Add(1)
    go func() {
        defer wg.Done()
        ....
    }()
}

wg.Wait()
close(results)

for result := range results {
  ...
}
```
It won't block because you've closed the channel after all goroutines have produced values.